### PR TITLE
[POR-2218] CLI changes to support datastore connect

### PR DESCRIPTION
**Pull request by ferozemohideen**
how it works:
1. get datastore row from the table
2. get the cluster to pull datastore credentials from
3. parse the datastore credentials according to the datastore type
4. use the credentials (host name, port) to create a proxy pod in the cluster from 2, which connects to the datastore
5. print out the credentials
6. portforward to the proxy pod

### CODE REVIEW

File: api/client/datastore.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/porter-dev/porter/api/types"
+)
+
+// GetDatastore gets a datastore by name for a project
+func (c *Client) GetDatastore(
```
Comment (d-g-town) [inline api/client/datastore.go:11]: nit: don't call this `Get`

File: cli/cmd/commands/datastore.go [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+	return forwardPorts("POST", req.URL(), config.RestConf, []string{fmt.Sprintf("%d:%d", port, datastoreCredential.Port)}, stopChannel, readyChannel)
+}
+
+type datastoreCredentialInput struct {
+	DatastoreName string
+	DatastoreType types.DatastoreType
+	Clientset     *kubernetes.Clientset
+}
+
+func datastoreCredential(inp datastoreCredentialInput) (types.DatastoreCredential, error) {
```
Comment (d-g-town) [inline cli/cmd/commands/datastore.go:286]: this should happen server-side

File: cli/cmd/commands/datastore.go [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+			},
+		},
+	}
+
+	s = spinner.New(spinner.CharSets[9], 100*time.Millisecond)
+	s.Color("cyan") // nolint:errcheck,gosec
+	s.Suffix = " Creating secure tunnel to datastore..."
+
+	s.Start()
+	proxyPod, err := config.Clientset.CoreV1().Pods(Namespace_Default).Create(context.TODO(), pod, metav1.CreateOptions{})
```
Comment (stefanmcshane) [inline cli/cmd/commands/datastore.go:235]: shouldnt use context.Todo

File: cli/cmd/commands/datastore.go [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+	datastoreConnectCmd := &cobra.Command{
+		Use:   "connect <DATASTORE_NAME>",
+		Short: "Forward a local port to a remote datastore.",
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			err := checkLoginAndRunWithConfig(cmd, cliConf, args, datastoreConnect)
+			if err != nil {
+				os.Exit(1)
+			}
+		},
```
Comment (d-g-town) [inline cli/cmd/commands/datastore.go:46]: consider using flags for datastore name instead of args

**Pull request merged**